R/markdown-link.R

```diff
@@ -53,21 +52,21 @@ add_linkrefs_to_md <- function(text) {
     )
   )[[1]]
 
-  if (length(refs) == 0) return(text)
+  if (length(refs) == 0) return("")
 
   ## For the [fun] form the link text is the same as the destination.
   # Need to check both NA and "" for different versions of stringr
   refs[, 3] <- ifelse(is.na(refs[,3]) | refs[,3] == "", refs[, 2], refs[,3])
 
   refs3encoded <- map_chr(refs[,3], URLencode)
   ref_text <- paste0("[", refs[, 3], "]: ", "R:", refs3encoded)
+  paste(ref_text, collapse = "\n")
+}
 
-  paste0(
-    text,
-    "\n\n",
-    paste(ref_text, collapse = "\n"),
-    "\n"
-  )
+add_linkrefs_to_md <- function(text) {
+  ref_text <- get_md_linkrefs(text)
+  if (! length(ref_text)) return(text)
```

Comment:
I suggest having `add_linkrefs_to_md()` return a character vector, since that will make testing a bit easier.  (Currently your check here is inconsistent with the early return above.)

Reply:
Perhaps you mean `get_md_linkrefs()`. (`add_linkrefs_to_md()` has the text as a string as input already.) Yeah that makes sense, it was like that in a previous version, hence the incorrect early return.... 

Done in b4eb16de0082ba003e35e57f2cf6894b7dd422d9